### PR TITLE
Normalize timespec borrow/overflow and add regression tests

### PR DESCRIPTION
**Pull request by Rtur2003**
## Summary                                                                                                        
- normalize timespec nsec overflow/borrow before conversions and expose helper for tests                          
- fix system_clock sec/nsec reconstruction drift                                                                  
- add regression tests for Time::toTimespec/fromTimespec and diff/normalize edge cases                            

## Testing                                                                                                        
- tests: ctest (gtest helpers/Time*, helpers/TimeDiff*)

### CODE REVIEW

File: scripts/generateShaderIncludes.sh [hunk 1]
```diff
@@ -1,24 +1,49 @@
 #!/bin/sh
 
+# dash-compatible and space-safe shader include generator. @Rtur2003
```
Comment (vaxerski) [inline scripts/generateShaderIncludes.sh:3]: what are these changes

File: src/helpers/time/Time.cpp [hunk 2]
```diff
@@ -137,3 +132,15 @@ struct timespec Time::toTimespec(const steady_tp& tp) {
     auto sum = timeadd(tpTime, diffFinal);
     return timespec{.tv_sec = sum.first, .tv_nsec = sum.second};
 }
+
+namespace Time::detail {
+
+sec_nsec diff(const sec_nsec& newer, const sec_nsec& older) {
```
Comment (vaxerski) [inline src/helpers/time/Time.cpp:138]: Time::detail::diff instead of namespace {}
